### PR TITLE
Desktop: Fix brightness configuration for OSTC4

### PR DESCRIPTION
**Pull request by mikeller**
### Describe the pull request:

- [x] Bug fix
- [ ] Functional change
- [ ] New feature
- [ ] Code cleanup
- [ ] Build system change
- [ ] Documentation change
- [ ] Language translation

### Pull request long description:

The Heinrichs Weikamp OSTC4 introduced an additional
'Cave' brightness level that is dimmer than all existing ones.

### Changes made:

1) 'Cave' level 'for the brightness combobox contents for OSTC4

### Release note:

Not sure of the desired level of detail covered in the changelog, but this is my proposal:
```desktop: fix brightness configuration for OSTC4```

### Mentions:

Signed-off-by: Michael Keller <github@ike.ch>

### CODE REVIEW

**Review by dirkhh — COMMENTED** [review]
This looks reasonable to me - but I don't have a way to verify this.
I had kinda hoped that @glance-  would comment as that is code he's much more involved with, but I can't see anything in here that seems broken.

Anyone else who wants to comment

(and yes, I am way behind on dealing with PRs... things have been a bit busy in real life)

**Comment by mikeller** [comment]
@dirkhh: Happy to do any specific testing for you if you let me know what you'd like to be tested. And don't worry about this being open for some time - I am not in a hurry to get this done. After all big part of a community driven open source project is that people do the work because they want to, and not because they have to.

**Comment by glance-** [comment]
Oh. I missed this because Jul and after that a week of Mine diving. I'll look at this in the coming days.

File: core/devicedetails.cpp [hunk 1]
```diff
@@ -13,7 +13,6 @@ setpoint::setpoint(unsigned char sp, unsigned char depth) :
 
 DeviceDetails::DeviceDetails(QObject *parent) :
 	QObject(parent),
-	data(0),
```
Comment (glance-) [inline core/devicedetails.cpp]: Put cleanup like this in a separate commit.

File: desktop-widgets/configuredivecomputerdialog.cpp [hunk 2]
```diff
@@ -264,7 +265,7 @@ void OstcFirmwareCheck::checkLatest(QWidget *_parent, device_data_t *data)
 {
 	devData = *data;
 	parent = _parent;
-	// If we didn't find a current firmware version stop this hole thing here.
+	// If we didn't find a current firmware version stop this whole thing here.
```
Comment (glance-) [inline desktop-widgets/configuredivecomputerdialog.cpp]: Also, with fixing spelling like this, put those spelling fixes in a separate commit.

File: desktop-widgets/configuredivecomputerdialog.cpp [hunk 3]
```diff
@@ -274,7 +275,7 @@ void OstcFirmwareCheck::checkLatest(QWidget *_parent, device_data_t *data)
 
 	int firmwareOnDevice = devData.devinfo.firmware;
 	QString firmwareOnDeviceString;
-	// Convert the latestFirmwareAvailable to a integear we can compare with
+	// Convert the latestFirmwareAvailable to a integer we can compare with
```
Comment (glance-) [inline desktop-widgets/configuredivecomputerdialog.cpp]: Same here. Put the spelling fixes in a separate commit.

File: desktop-widgets/configuredivecomputerdialog.cpp [hunk 4]
```diff
@@ -1242,11 +1243,28 @@ void ConfigureDiveComputerDialog::reloadValuesSuuntoVyper()
...
+		QStandardItem* item = (qobject_cast<QStandardItemModel*>(ui.brightnessComboBox_4->model()))->item(3);
+		item->setFlags(item->flags() & Qt::ItemIsEnabled);
+	} else {
+		ui.brightnessComboBox_4->setItemText(0, tr("Eco"));
+		ui.brightnessComboBox_4->setItemText(1, tr("Medium"));
+		ui.brightnessComboBox_4->setItemText(2, tr("High"));
+		(qobject_cast<QListView *>(ui.brightnessComboBox_4->view()))->setRowHidden(3, true);
+		QStandardItem* item = (qobject_cast<QStandardItemModel*>(ui.brightnessComboBox_4->model()))->item(3);
+		item->setFlags(item->flags() & ~Qt::ItemIsEnabled);
+	}
```
Comment (glance-) [inline desktop-widgets/configuredivecomputerdialog.cpp]: I haven't cared much about compatibility with anything except the last/later versions in any of these codebases. 1.6.0 is about 5 months old, so I hope most people have upgraded by now.

Btw. From what I see it the Cave setting has been there since 1.3.0.

So, I'd suggest the whole firmware version magic and just adding the Cave entry to the combo box,.

**Review by glance- — CHANGES_REQUESTED** [review]
Please drop the firmware version check. My research in the OSTC4 code tells me the setting has bin tere since 1.3.0 which is 2018, and I haven't had any expectation that this code supports anything but the latest / late firmwares.

Break out spelling fixes and unrelated code cleanup to separate commits.

**Review by glance- — APPROVED** [review]

**Comment by bstoeger** [comment]
Great, I think we can merge.
